src/JsonSchema/Constraints/Factory.php

```diff
@@ -41,15 +46,17 @@ class Factory
     );
 
     /**
+     * @param SchemaStorage $schemaStorage
      * @param UriRetriever $uriRetriever
      */
-    public function __construct(UriRetriever $uriRetriever = null)
+    public function __construct(SchemaStorage $schemaStorage = null, UriRetriever $uriRetriever = null)
     {
         if (!$uriRetriever) {
             $uriRetriever = new UriRetriever();
         }
 
         $this->uriRetriever = $uriRetriever;
+        $this->schemaStorage = $schemaStorage;
```

Comment:
since `$schemaStorage` defaults to `null` - shouldn't we initiate a default `SchemaStorage` instance?